src/docs/development/add-to-app/android/plugin-setup.md

```diff
@@ -0,0 +1,128 @@
+---
+title: Managing plugins and dependencies in add-to-app
+short-title: Plugin setup
+description: Learn how to use plugins and share your plugin's library dependencies with your existing app.
+---
+
+This guide describes how to set up your project to consume plugins and how to
+manage your Gradle library dependencies between your existing Android app and
+your Flutter module's plugins.
+
+## A. Simple scenario
+
+In the simple cases:
+
+- Your Flutter module uses a plugin that has no additional Android Gradle
+  dependency because it only uses Android OS APIs, such as the camera plugin.
+- Your Flutter module uses a plugin that has an Android Gradle dependency,
+  such as [ExoPlayer from the video_player plugin](https://github.com/flutter/plugins/blob/master/packages/video_player/video_player/android/build.gradle),
+  but your existing Android app didn't depend on ExoPlayer.
+
+There are no additional steps needed. Your add-to-app module will work the same
+way as a full-Flutter app. Whether you integrate via Android Studio, via
+Gradle subproject or using AARs, transitive Android Gradle libraries will be
+automatically bundled as needed into your outer existing app.
+
+## B. Plugins needing project edits
+
+Some plugins require you to make some edits to the Android side of your project.
+
+Taking the [firebase_crashlytics plugin](https://pub.dev/packages/firebase_crashlytics)
+as an example. The plugin's integration instructions requires some manual edits
+to your Android wrapper project's `build.gradle` file.
+
+In the case of a Flutter module, which only has Dart files, perform those
```

Comment:
Make these Gradle file edits on the outer, existing Android app rather than in your Flutter module, which only contains dart files.